utils.js

```diff
@@ -43,10 +43,66 @@ const prefs = require('sdk/simple-prefs');
 const {get, set, reset} = require('sdk/preferences/service');
 const NS_XUL = 'http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul';
 const self = require('sdk/self');
+const main = require('main');
 const {studyConfig} = require('shield-config.js');
+const {AddonManager} = require('resource://gre/modules/AddonManager.jsm');
 
 const shield = require('shield-studies-addon-utils');
-const TCStudy = new shield.Study(studyConfig);
+const INCOMPATIBLE_ADDONS = {
+  'TooManyTabs@visibotech.com': 'Too Many Tabs',
+  'firebug@software.joehewitt.com': 'Firebug',
+  'hidecaptionplus-dp@dummy.addons.mozilla.org': 'Hide Caption Titlebar Plus',
+  'tabgroups@quicksaver': 'Tab Groups',
+  'tabgroupshelper@kevinallasso.org': 'Tab Groups Helper',
+  'treestyletab@piro.sakura.ne.jp': 'Tree-Style Tab',
+  '{097d3191-e6fa-4728-9826-b533d755359d}': 'All-In-One Sidebar',
+  '{c6448328-31f7-4b12-a2e0-5c39d0290307}': 'HTitle',
+  '{dc572301-7619-498c-a57d-39143191b318}': 'Tab Mix Plus',
+  'multipletab@piro.sakura.ne.jp': 'Multiple Tab Handler',
+  'tabscope@xuldev.org': 'Tab Scope',
+  'bug566510@vovcacik.addons.mozilla.org': 'Allow multiselect operations on tabs',
+  'jid1-rOGUyxs2TvBBuA@jetpack': 'Previous tab highlighting',
+  '{fc2b8f80-d9a5-4f51-8076-7c7ce3c67ee3}': 'Diigo Toolbar',
+  'tabcentertest1@mozilla.com': 'Tab Center',
+  '{ec8030f7-c20a-464f-9b0e-13a3a9e97384}': 'Tab Tree',
+  '@testpilot-addon': 'Test Pilot',
+  '{0545b830-f0aa-4d7e-8820-50a4629a56fe}': 'ColorfulTabs',
+  'rightbar@realmtech.net': 'RightBar',
+  'tabsonbottom@piro.sakura.ne.jp':'Tabs on Bottom',
+  'extension@one-tab.com':'OneTab',
+  'bartablitex@szabolcs.hubai':'BarTab Lite X',
+  'jid0-AjzBVlpzVAaBqxcar9QDqMWWAVQ@jetpack':'Side Tabs',
+  'TabsTree@traxium':'Tab Tree',
+  '{e5bbc237-c99b-4ced-a061-0be27703295f}':'Hide Tab Bar With One Tab',
+  'tiletabs@DW-dev':'Tile Tabs',
+  'bartabheavy@philikon.de':'BarTab Heavy',
+};
+
+class tabCenterStudy extends shield.Study {
+  constructor(config) {
+    super(config);
+  }
+
+  isEligible() {
```

Comment:
so, ways of testing this... you could 'fake it', by amending the arity to `isEligible(addonStruct=[])`, then doing the thing, or pulling the 'hasAConflictingAddon' part out, and calling it with your list.  This changes the problem to 'how well can I shim a convincing telemetry addon list', which might or might not be a simpler problem.